goalie/go_mesh_seq.py

```diff
@@ -87,54 +91,89 @@ def get_enriched_mesh_seq(
         """
         if not hasattr(self, "_indicators"):
             self._create_indicators()
         return self._indicators
 
     @PETSc.Log.EventDecorator()
     def indicate_errors(
-        self,
-        enrichment_kwargs: dict = {},
-        adj_kwargs: dict = {},
-        indicator_fn: Callable = get_dwr_indicator,
-    ) -> Tuple[dict, AttrDict]:
+        self, enrichment_kwargs={}, solver_kwargs={}, indicator_fn=get_dwr_indicator
+    ):
         """
         Compute goal-oriented error indicators for each subinterval based on solving the
         adjoint problem in a globally enriched space.
 
         :kwarg enrichment_kwargs: keyword arguments to pass to the global enrichment
-            method
-        :kwarg adj_kwargs: keyword arguments to pass to the adjoint solver
-        :kwarg indicator_fn: function for error indication, which takes the form,
-            adjoint error and enriched space(s) as arguments
+            method - see :meth:`~.GoalOrientedMeshSeq.get_enriched_mesh_seq` for the
+            supported enrichment methods and options
+        :type enrichment_kwargs: :class:`dict` with :class:`str` keys and values which
+            may take various types
+        :kwarg solver_kwargs: parameters for the forward solver, as well as any
+            parameters for the QoI, which should be included as a sub-dictionary with key
+            'qoi_kwargs'
+        :type solver_kwargs: :class:`dict` with :class:`str` keys and values which may
+            take various types
+        :kwarg indicator_fn: function which maps the form, adjoint error and enriched
+            space(s) as arguments to the error indicator
+            :class:`firedrake.function.Function`
+        :returns: solution and indicator data objects
+        :rtype1: :class:`~.AdjointSolutionData
+        :rtype2: :class:`~.IndicatorData
         """
         enrichment_kwargs.setdefault("enrichment_method", "p")
         enrichment_kwargs.setdefault("num_enrichments", 1)
         enriched_mesh_seq = self.get_enriched_mesh_seq(**enrichment_kwargs)
         transfer = self._get_transfer_function(enrichment_kwargs["enrichment_method"])
 
+        # Reinitialise the error indicator data object
+        self._create_indicators()
+
         # Solve the forward and adjoint problems on the MeshSeq and its enriched version
-        self.solve_adjoint(**adj_kwargs)
-        enriched_mesh_seq.solve_adjoint(**adj_kwargs)
+        self.solve_adjoint(**solver_kwargs)
+        enriched_mesh_seq.solve_adjoint(**solver_kwargs)
+
+        tp = self.time_partition
+        # check whether get_form contains a kwarg indicating time-dependent constants
+        timedep_const = "err_ind_time" in signature(enriched_mesh_seq.form).parameters
+
+        tp = self.time_partition
+        # check whether get_form contains a kwarg indicating time-dependent constants
+        timedep_const = "err_ind_time" in signature(enriched_mesh_seq.form).parameters
```

Comment:
Not sure what happened here with the duplication. But - in either case - we can get rid of the `signature` stuff now that `err_ind_time` has been dropped.

Reply:
Done in 29e959f5b82f28c496433347c3492008a868354c